<?xml version="1.0" ?>
<launch>
  <arg name="vehicle" default="$(env ROS_HOSTNAME)" />
  <arg name="ns" default="/$(arg vehicle)" />
  <!-- Type of node.  Current options are 'robot' and 'base' -->
  <arg name="type" default="beacon" />
  <!-- Merger type - 0: Octomap, 1: PCL-Kyle, 2: PCL-Jessup -->
  <arg name="merger" default="0" />
  <!-- Whether to merge everything (true) or prioritize own map (false) -->
  <arg name="full_merge" default="true" />
  <!-- Whether to prioritize free space from any agent -->
  <arg name="free_prioritize" default="false" />
  <!-- Octomap type - 0: Binary, 1: Full -->
  <arg name="octoType" default="0" />
  <!-- Map resolution.  If different res needed for each need to change code -->
  <arg name="resolution" default="0.2" />
  <!-- Rate to run the node at -->
  <arg name="rate" default="0.5" />
  <!-- Size of map differences to trigger a merge -->
  <arg name="mapThresh" default="100" />
  <!-- Topics to subscribe and publish -->
  <arg name="mapTopic" default="octomap_binary" />
  <arg name="neighborsTopic" default="neighbor_maps" />
  <arg name="mergedTopic" default="merged_map" />
  <arg name="mergedSizeTopic" default="merged_size" />
  <arg name="pclTopic" default="pc2_out" />

  <node ns="$(arg ns)" name="octomap_merger" pkg="octomap_merger" type="octomap_merger_node" output="screen">
    <param name="vehicle" value="$(arg vehicle)" />
    <param name="type" value="$(arg type)" />
    <param name="merger" value="$(arg merger)" />
    <param name="full_merge" value="$(arg full_merge)" />
    <param name="free_prioritize" value="$(arg free_prioritize)" />
    <param name="octoType" value="$(arg octoType)" />
    <param name="resolution" value="$(arg resolution)" />
    <param name="rate" value="$(arg rate)" />
    <param name="mapThresh" value="$(arg mapThresh)" />
    <param name="mapTopic" value="$(arg mapTopic)" />
    <param name="neighborsTopic" value="$(arg neighborsTopic)" />
    <param name="mergedTopic" value="$(arg mergedTopic)" />
    <param name="mergedSizeTopic" value="$(arg mergedSizeTopic)" />
    <param name="pclTopic" value="$(arg pclTopic)" />
  </node>
</launch>
